File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l, r = 0, len(numbers) - 1

        while l < r:
            small_num, big_num = numbers[l], numbers[r]

            if small_num + big_num < target:
                l += 1
            elif small_num + big_num > target:
                r -= 1
            else:
                # assert small_num + big_num == target
                return [l + 1, r + 1]
        
        raise Exception("Unreachable Code")

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int left = 0, right = numbers.Length - 1;

        while (numbers[left] + numbers[right] != target)
        {
            if (numbers[left] + numbers[right] > target)
                right--;
            else
                left++;
        }

        // Return the indices of the two numbers, index1 and index2, 
        // added by one as an integer array [index1, index2] 
        // of length 2.
        left++; right++;
        return left < right ? new int[] {left, right} : new int[] {right, left};
    }
}